File: CardViewExample/app/src/main/java/com/hfad/cardviewexample/MainActivity.java
package com.hfad.cardviewexample;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.ImageView;
import android.widget.ListView;
import android.widget.TextView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";
    private ListView mListView;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.listview_layout);
        mListView = (ListView) findViewById(R.id.listView);

        ArrayList<Card> list = new ArrayList<>();
        list.add(new Card("@drawable://" + R.drawable.a1, "A1"));
        list.add(new Card("@drawable://" + R.drawable.a2, "A2"));
        list.add(new Card("@drawable://" + R.drawable.a3, "A3"));
        list.add(new Card("@drawable://" + R.drawable.a4, "A4"));
        list.add(new Card("@drawable://" + R.drawable.a5, "A5"));
        list.add(new Card("@drawable://" + R.drawable.a6, "A6"));
        list.add(new Card("@drawable://" + R.drawable.a7, "A7"));
        list.add(new Card("@drawable://" + R.drawable.a8, "A8"));
        list.add(new Card("@drawable://" + R.drawable.a9, "A9"));
        list.add(new Card("@drawable://" + R.drawable.a10, "A10"));

        CustomListAdapter adapter = new CustomListAdapter(this, R.layout.activity_main, list);
        mListView.setAdapter(adapter);

    }
}

File: SemaforKotlin/app/src/main/java/com/hfad/semaforkotlin/MainActivity.kt
package com.hfad.semaforkotlin

import android.app.Activity
import android.os.Bundle
import android.view.View
import android.widget.ImageButton
import android.widget.ImageView

class MainActivity : Activity() {

    var imSemafor: ImageView? = null
    var imButton: ImageButton? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        imSemafor = findViewById(R.id.ivSemafor)
        imButton = findViewById(R.id.ivButton)

    }

    fun onClickStartStop(view: View) {

        imSemafor?.setImageResource(R.drawable.semafor_green)
        imButton?.setImageResource(R.drawable.button_stop)
    }
}
